lib/kb_Metrics/metricsDBs.py

```diff
@@ -87,133 +84,124 @@ def update_activity_records(self, upd_filter, upd_data):
             # return an instance of UpdateResult(raw_result, acknowledged)
             update_ret = self.mt_coll.update_one(upd_filter,
                                                  upd_op, upsert=True)
-        except BulkWriteError as bwe:
-            # pprint(bwe.details['writeErrors'])
+        except WriteError as we:
+            # pprint(we.details['writeErrors'])
             panic = filter(lambda x: x['code'] != 11000,
-                           bwe.details['writeErrors'])
+                           we.details['writeErrors'])
             if len(panic) > 0:
```

Comment:
Also again, I am not quite following you why you want to raise with a condition. My point is you catch an 'WriteError' exception and raise only when panic > 0. What if your code run into WriteError and panic < 0? In this case, you don't raise anything? 

Please add a test for this to explain for future developers.

Reply:
Because 'WriteError & panic <=0" means when error_code is 11000, and that is okay to skip.  If the key is already existing, no insertion is performed yet I want the process to keep going to the next record instead of stall.

Reply:
okay. Please leave a comment then on your code.

Reply:
fixed--mixed with insert_many before.